cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -79,13 +87,53 @@
   @ClassRule
   public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
 
+  private static NamespaceId configuredNamespace = configureTestNamespace();
+  // boolean is true if we created the namespace; indicates that we are responsible for cleaning it
+  private static Map<NamespaceId, Boolean> registeredNamespaces = new HashMap<>();
+
   private AccessToken accessToken;
 
   @Before
   public void setUp() throws Exception {
     checkSystemServices();
     assertUnrecoverableResetEnabled();
-    assertIsClear();
+
+    boolean createdByUs = false;
+    if (!getNamespaceClient().exists(configuredNamespace.toId())) {
```

Comment:
So in that case, in a CI environment, do we expect that a user should set up the namespace in a wrapper script, and then start integration tests?

Should we instead allow users to specify those namespace configurations as well, and have the framework manage the namespace?

Not leaning one way or the other, but something to consider.